src/Fixer/Casing/MagicMethodCasingFixer.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Casing;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class MagicMethodCasingFixer extends AbstractFixer
+{
+    private static $magicNames = [
+        '__call' => '__call',
+        '__callstatic' => '__callStatic',
+        '__clone' => '__clone',
+        '__construct' => '__construct',
+        '__debuginfo' => '__debugInfo',
+        '__destruct' => '__destruct',
+        '__get' => '__get',
+        '__invoke' => '__invoke',
+        '__isset' => '__isset',
+        '__set' => '__set',
+        '__set_state' => '__set_state',
+        '__sleep' => '__sleep',
+        '__tostring' => '__toString',
+        '__unset' => '__unset',
+        '__wakeup' => '__wakeup',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Magic method definitions and calls must be using the correct casing.',
+            [
+                new CodeSample(
+'<?php
+class Foo
+{
+    public function __Sleep()
+    {
+    }
+}
+'
+                ),
+                new CodeSample(
+'<?php
+$foo->__INVOKE(1);
+'
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_STRING);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $inClass = 0;
+        $tokenCount = count($tokens);
+
+        for ($index = 1; $index < $tokenCount - 2; ++$index) {
+            if (0 === $inClass && $tokens[$index]->isClassy()) {
+                $inClass = 1;
```

Comment:
Would it be easier to make this variable be `null` by default and then store the result of `$tokens->findBlockEnd()` so that you just compare it to the current index instead of couting `{` and `}` tokens? The variable name could become e.g. `$inClassUntil`.

Reply:
That would require scanning from the classy open `{` till the end and than go back scanning all the tokens again for candidates, which is not very efficient.